### PR TITLE
Use isinstance for comparing types (E721)

### PR DESCRIPTION
**Pull request by winniehell**
## Description of the issue/feature this PR addresses

Enables check for using `isinstance` when comparing types (E721) and fixes existing violations.

### CODE REVIEW

File: src/senaite/core/browser/fields/record.py [hunk 1]
```diff
@@ -246,7 +247,7 @@ def set(self, instance, value, **kwargs):
         ObjectField.set(self, instance, value, **kwargs)
 
     def _to_dict(self, value):
-        if type(value) != type({}) and hasattr(value, 'keys'):
+        if not isinstance(value, dict) and hasattr(value, 'keys'):
             new_value = {}
             new_value.update(value)
             return new_value
```
Comment (winniehell) [inline src/senaite/core/browser/fields/record.py:249]: I'd like to leave a comment on the next line (253/254) but GitHub won't let me. 🙄

Do we really want to return `value` here if it is not an instance of `dict`?

Comment (ramonski) [reply src/senaite/core/browser/fields/record.py:249]: Good question and no idea at the moment. Anyways, these Archetype fields will be all removed as soon as we have migrated all contents/fields to Dexterity.

**Review by ramonski — APPROVED** [review]

**Review by ramonski — APPROVED** [review]

File: src/senaite/core/browser/fields/record.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 # Copyright 2018-2021 by it's authors.
 # Some rights reserved, see README and LICENSE.
 
+import six
+from types import ClassType
```
Comment (winniehell) [inline src/senaite/core/browser/fields/record.py:22]: this import is unused and will make flake8 complain—I'll take care of this

Comment (winniehell) [reply src/senaite/core/browser/fields/record.py:22]: fixed in https://github.com/senaite/senaite.core/pull/2091

**Pull request merged**